### DISCUSSION TITLE
How to add tracks to MsaView

### DISCUSSION THREAD

**Discussion by mamrosh** [GMOD/jbrowse-components#4144] [General]
Hi, I haven't been able to figure out how to add tracks to MsaView (on my server). I can add them to LGV, but nothing ever shows up for MsaView. However, I see tracks of text ("Sequence consensus", "Secondary-structure") on the MsaView demos (above the alignment). I would like to add something similar and am wondering how. 

Thanks! Jennifer

**Comment by cmdcolin** [comment]
This is a good question, and I don't have a great answer, but basically the MsaView plugin does not use the same notion of "tracks" that the normal genome browser tracks in jbrowse have. 

The MsaView has some specialized notions of special tracks. For example, the secondary-structure and sequence consensus tracks in the demos are data that is contained in the alignment files themselves (the .stock stockholm format files for our demo had the secondary structure data, and clustal format files have the sequence consensus with the '*' letters on certain columns), so that is just data in the file formats we loaded, not separate track files, and it's not hooked up the the normal jbrowse track selector. There are also some special cases in the MsaView app where you can also add protein domain annotations in GFF from uniprot. 

Anyways, all this to say is: the MsaView actually does not currently have a super general notion of tracks. Part of this was because we created the MsaView app as a standalone thing and added to jbrowse as a second step (standalone app here https://gmod.github.io/react-msaview/) and part of this was because the notion of tracks did not cleanly map over to the MSA coordinate space the same way as tracks on a reference genome works! But, I'd be happy to help with your use case, if you are able to describe your use case I would be happy to try to advise


Also, we are working on an MsaView paper, so I am definitely happy to assist in making it more useful to others :)
